sandy/core/cov.py

```diff
@@ -399,14 +406,39 @@ def glls(self, S, Vy):
         C = Vx - Vx.dot(S.dot(np.linalg.inv(V).dot(S.T.dot(Vx))))
         return self.__class__(C)
 
-    def sandwich(self, Si, Sj=None):
+    def sandwich(self, S):
+        """
+        Apply the sandwich formula to the CategoryCov object for a given
+        pandas.Series.
+
+        Parameters
+        ----------
+        S : `pd.Series`
+            Standard desviations.
+
+        Returns
+        -------
+        `CategoryCov`
+            `CategoryCov` object to which we have applied sandwich
+            formula for a given pd.Series
+
+        Examples
+        --------
+        >>> S = np.diag(np.array([1,2,3]))
+        >>> var = pd.Series([1, 2, 3])
+        >>> cov = sandy.CategoryCov(S)
+        >>> cov.sandwich(var)
+                    0           1           2
+        0 1.00000e+00 0.00000e+00 0.00000e+00
+        1 0.00000e+00 8.00000e+00 0.00000e+00
+        2 0.00000e+00 0.00000e+00 2.70000e+01
+        """
+        if self.data.index.values.all() != S.index.values.all():
+            raise sandy.Error("The index are not the same")

```

Comment:
"the indices of the sensitivity array and of the `CategoryCov` are not the same"